src/harness/unittests/tsserverProjectSystem.ts

```diff
@@ -254,33 +254,34 @@ namespace ts.projectSystem {
     }
 
     export function isFolder(s: FSEntry): s is Folder {
-        return isArray((<Folder>s).entries);
+        return s && isArray((<Folder>s).entries);
     }
 
     export function isFile(s: FSEntry): s is File {
-        return typeof (<File>s).content === "string";
+        return s && typeof (<File>s).content === "string";
     }
 
-    export function addFolder(fullPath: string, toPath: (s: string) => Path, fs: Map<FSEntry>): Folder {
-        const path = toPath(fullPath);
-        if (fs.has(path)) {
-            Debug.assert(isFolder(fs.get(path)));
-            return (<Folder>fs.get(path));
+    function invokeDirectoryWatcher(callbacks: DirectoryWatcherCallback[], getRelativeFilePath: () => string) {
+        if (callbacks) {
+            const cbs = callbacks.slice();

```

Comment:
This is to ensure that if the watching callbacks are removed when they are called into does not miss any of the callbacks following it